### PR TITLE
[runtime/curdir] implement gethostname

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: runtime/flang/curdir.c [hunk 1]
```diff
@@ -115,8 +115,12 @@ void __fort_gethostname(host) char *host;
     }
     p = un.nodename;
   }
-  strcpy(host, p);
 #else
-  strcpy(host, "localhost");
-#endif	  
+  #include <Winsock2.h>
+  s = gethostname(&p, 256);
```
Comment (isuruf) [inline runtime/flang/curdir.c]: this needs `Ws2_32.lib` right?

Comment (ghost) [reply runtime/flang/curdir.c]: Yeah there are few libraries that are included. I'll toss that in.

**Comment by ghost** [comment]
Also, will include the `if(WIN32)` on resolving the merge conflict.

File: CMakeLists.txt [hunk 2]
```diff
@@ -341,9 +341,9 @@ macro(add_flang_library name)
     add_dependencies( ${name} ${LLVM_COMMON_DEPENDS} )
   endif( LLVM_COMMON_DEPENDS )
 
-  llvm_config( ${name} ${LLVM_LINK_COMPONENTS} )
+  llvm_config( ${name} ${LLVM_LINK_COMPONENTS}
```
Comment (isuruf) [inline CMakeLists.txt]: `)` is missing here

File: CMakeLists.txt [hunk 3]
```diff
@@ -341,9 +341,9 @@ macro(add_flang_library name)
     add_dependencies( ${name} ${LLVM_COMMON_DEPENDS} )
   endif( LLVM_COMMON_DEPENDS )
 
-  llvm_config( ${name} ${LLVM_LINK_COMPONENTS} )
+  llvm_config( ${name} ${LLVM_LINK_COMPONENTS} 
   if(WIN32)
-    list(APPEND LLVM_COMMON_LIBS Dbghelp.lib)
+    list(APPEND LLVM_COMMON_LIBS Dbghelp.lib Ws2_32.lib)
```
Comment (isuruf) [inline CMakeLists.txt]: I'm wondering this is the correct place for this. How about `runtime/flang/CMakeLists.txt`?

Comment (ghost) [reply CMakeLists.txt]: It's not the correct place, but I'm cmake-impaired.

Comment (ghost) [reply CMakeLists.txt]: Do you have a line number or would you like to fix it?

Comment (isuruf) [reply CMakeLists.txt]: https://github.com/isuruf/flang/blob/windows/runtime/flang/CMakeLists.txt#L594 something like
```
target_link_libraries(flang_shared PRIVATE Ws2_32.lib)
target_link_libraries(flang_static PRIVATE Ws2_32.lib)
```

**Pull request merged**